pandas/tests/frame/methods/test_asfreq.py

```diff
@@ -91,3 +91,16 @@ def test_asfreq_with_date_object_index(self, frame_or_series):
         result = ts2.asfreq("4H", method="ffill")
         expected = ts.asfreq("4H", method="ffill")
         tm.assert_equal(result, expected)
+
+    def test_asfreq_with_unsorted_index(self, frame_or_series):
+        # GH#39805
+        # Test that rows are not dropped when the datetime index is out of order
+        index = to_datetime(["2021-01-04", "2021-01-02", "2021-01-03", "2021-01-01"])
+        ts = frame_or_series(range(4), index=index)
+
+        sorted_index = to_datetime(
+            ["2021-01-01", "2021-01-02", "2021-01-03", "2021-01-04"]
+        )
+        expected = frame_or_series([3, 1, 2, 0], index=sorted_index)
+
+        tm.assert_equal(ts.asfreq("D"), expected.asfreq("D"))
```

Comment:
why do you need to asfreq these?

Reply:
I asfreq `ts` (`result`) because that's what we're testing. I need to asfreq `expected` because tm.assert_equal will complain otherwise. I'll add a comment with an explanation.